syntax = "proto3";

package nglm_grpc;

service Server {
	rpc register(clientInfo) returns (registerResponse) {}
	rpc isAlive(query) returns (response) {}
}

message clientInfo {
	string hostname = 1;
	string ipv4 = 2;
	int32 port = 3;
	string uuid = 4;
}

message registerResponse {
	string uuid = 1;
	bool success = 2;
}

message response {
	bool success = 1;
}

message query {
	string query = 1;
}

service Logging {
	rpc start(params) returns (response) {}
	rpc output(stream chunks) returns (response) {}
	rpc err(exception) returns (response) {}
	rpc getConfig(chunkSize) returns (stream chunks) {}
	rpc setConfig(stream chunks) returns (response) {}
}

message exception {
	string exception = 1;
}

message chunkSize {
	double size = 1;
}

message params {
	int32 pid = 1;
	string pname = 2;
	int32 interval = 3;
	int32 duration = 4;
	string taskUUID = 5;
}

message chunks {
	bytes buffer = 1;
}